src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder getSalaryOutput = new StringBuilder();
+        int[] salary = new int[names.length];
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        final LocalDate dateFromFormated = LocalDate.parse(dateFrom, formatter);
+        final LocalDate dateToFormated = LocalDate.parse(dateTo, formatter);
+        getSalaryOutput.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String dataArray : data) {
+            String[] dataSplit = dataArray.split(" ");
+            LocalDate localDate = LocalDate.parse(dataSplit[0], formatter);
+            if (localDate.isAfter(dateFromFormated)
+                    && localDate.isBefore(dateToFormated)
+                    || localDate.isEqual(dateFromFormated)
+                    || localDate.isEqual(dateToFormated)) {
```

Comment:
The condition to check if the date is within the range can be simplified by using !localDate.isBefore(dateFromFormated) && !localDate.isAfter(dateToFormated), which is more readable.